test_AquariusError fails when run in isolation

# Description
`test_exceptions.py::test_AquariusError` fails when run in isolation.  

# Expected Behavior
Any given test should work when run in isolation and shouldn't depend on any other test.

# Error Log
```console
(venv) david@david-Inspiron-5502:~/projects/ocean.py$ pytest ocean_lib/test/test_exceptions.py::test_AquariusError
========================================================================== test session starts ===========================================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/david/projects/ocean.py, configfile: pytest.ini
plugins: env-0.6.2
collected 1 item                                                                                                                                                         

ocean_lib/test/test_exceptions.py F                                                                                                                                [100%]

================================================================================ FAILURES ================================================================================
___________________________________________________________________________ test_AquariusError ___________________________________________________________________________

publisher_ocean_instance = <ocean_lib.ocean.ocean.Ocean object at 0x7f1112c94a00>
metadata = {'main': {'author': 'Mario', 'dateCreated': '2012-02-01T10:55:11Z', 'files': [{'checksum': '62b254af-c001-4785-89c8-5a...om/datacommons-seeding-us-east/10_Monkey_Species_Small/assets/validation.zip'}], 'license': 'CC0: Public Domain', ...}}

    def test_AquariusError(publisher_ocean_instance, metadata):
        metadata_copy = metadata.copy()
        publisher = get_publisher_wallet()
    
>       did = publisher_ocean_instance.assets._get_aquarius().list_assets()[0]
E       IndexError: list index out of range

ocean_lib/test/test_exceptions.py:49: IndexError
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
sender: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58, 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260, None, address: 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260
private key: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58
public key: 0x8b1b222564fb8302e9acfa4bcd60060de03ae479fcbed38fe21e45cd39ddb3fefb60ba1b4625e00d02223027c8f485f6d77693ab62ce06289b23942c6b319ccf

sender balance: 91.772531962
======================================================================== short test summary info =========================================================================
FAILED ocean_lib/test/test_exceptions.py::test_AquariusError - IndexError: list index out of range
=========================================================================== 1 failed in 20.16s ===========================================================================
```

# Analysis
test_AquariusError relies on previous tests creating assets.  It should create its own assets.

--- a/ocean_lib/test/test_exceptions.py
+++ b/ocean_lib/test/test_exceptions.py
@@ -10,7 +10,7 @@ from eth_utils import add_0x_prefix
 from ocean_lib.common.ddo.ddo import DDO
 from ocean_lib.exceptions import InsufficientBalance, ContractNotFound, AquariusError
 from ocean_lib.web3_internal.constants import ZERO_ADDRESS
-from tests.resources.ddo_helpers import get_resource_path
+from tests.resources.ddo_helpers import get_resource_path, wait_for_ddo
 from tests.resources.helper_functions import get_publisher_wallet
 
 
@@ -46,7 +46,16 @@ def test_AquariusError(publisher_ocean_instance, metadata):
     metadata_copy = metadata.copy()
     publisher = get_publisher_wallet()
 
-    did = publisher_ocean_instance.assets._get_aquarius().list_assets()[0]
+    all_assets = publisher_ocean_instance.assets._get_aquarius().list_assets()
+
+    if not all_assets:
+        # the test is run in isolation, docker has been pruned,
+        # or Aqua is empty for some other reason
+        ddo = publisher_ocean_instance.assets.create(metadata_copy, publisher)
+        wait_for_ddo(publisher_ocean_instance, ddo.did)
+        all_assets = publisher_ocean_instance.assets._get_aquarius().list_assets()
+
+    did = all_assets[0]
     token_address = add_0x_prefix(did[7:])
 
     with pytest.raises(AquariusError):

